firebase-installations/REQUIRED_FIREBASE_OPTIONS_ANDROID.md

```diff
@@ -0,0 +1,38 @@
+  Example value: `1:1234567890:android:321abc456def7890`
+
+## What do I need to know?
+To improve security Firebase SDK updates [on February 27](https://firebase.google.com/support/release-notes/android#2020-02-27) and afterwards replaced the Firebase Instance ID service with a dependency on the [Firebase Installations API](https://console.cloud.google.com/apis/library/firebaseinstallations.googleapis.com).
+
+Firebase Installations enforces the existence and validity of mandatory Firebase options API key, Project ID, and Application ID in order to associate client data with your Firebase project. 
+## Firebase Cloud Messaging (FCM) with Firebase Instance ID (IID)
+If you are reading this message, most likely your application is initializing Firebase without the required set of Firebase options.
+
+Your application may be using an incomplete or erroneous [`google-services.json`](https://firebase.google.com/docs/reference/android/com/google/firebase/FirebaseApp) configuration file; or your app is [programmatically initializing Firebase](https://firebase.google.com/docs/projects/multiprojects) without the full set of required Firebase options.
+
+As a result, Firebase services like Firebase Cloud Messaging (FCM) will malfunction for end-users who installed your app after it was released with the updated Firebase SDKs. Additionally, repeated failing requests to Firebase may slow down the end-user experience of your app.
+
+## What do I need to do?
+To fix malfunctioning Firebase services for your applications, **please take the following steps as soon as possible:**
+1. Update your application by initializing Firebase with a valid API key of your project, a valid Project ID, and a valid Application ID (a.k.a. mobilesdk_app_id, GOOGLE_APP_ID, or simply "App Id").
+    * **Default initialization process using a Firebase config file**: [Download your `google-services.json` config file](https://support.google.com/firebase/answer/7015592) from the Firebase console, then replace the existing file in your app.
+    * **Programmatic initialization using a `FirebaseOptions` object**: [Download your `google-services.json` config file](https://support.google.com/firebase/answer/7015592) from the Firebase console to find your API key, Project ID, and Application ID, then update these values in the `FirebaseOptions` object in your app.
+1. Release a new version of your app to the Play Store.
+
+## FCM Server keys
+If your app is using an [FCM Server key](https://firebase.google.com/docs/cloud-messaging/auth-server#authorize-legacy-protocol-send-requests) rather than a Cloud API key, this could cause a security vulnerability in case you are using the same FCM Server key to send push notifications via FCM. \ In this case, we strongly recommend that you revisit how your server [authenticates send requests to FCM.](https://firebase.google.com/docs/cloud-messaging/auth-server)
```

Comment:
```suggestion
If your app is using an [FCM Server key](https://firebase.google.com/docs/cloud-messaging/auth-server#authorize-legacy-protocol-send-requests) rather than a Cloud API key, this could cause a security vulnerability in case you are using the same FCM Server key to send push notifications via FCM. In this case, we strongly recommend that you revisit how your server [authenticates send requests to FCM.](https://firebase.google.com/docs/cloud-messaging/auth-server)
```